pyrokinetics/gk_code/GKInputGENE.py

```diff
@@ -120,8 +129,6 @@ def get_local_geometry_miller(self) -> LocalGeometryMiller:
 
```

Comment:
Here we are reading in the data and working out the pyro normalisation. This should be fine when we are writing out new files and running new simulations. But if we wanted to load in an existing simulations that has `lref=major_radius` (which corresponds to `self.data["geometry"]["major_r"] = 1.0` then pyro wouldn't know how to normalise the outputs. I think when we add in the normalisation for the outputs we need to check the input file to see which normalisation was used.

Reply:
I was a bit confused by this -- if I understand correctly, GENE input files always (sometimes?) have both `major_r` _and_ `minor_r` set, and then Pyrokinetics will renormalise it so that `major_r == 1.0`? I don't have access to the GENE manual -- are these in metres, or normalised units?

With this PR, reading in any GENE file will result in lengths being defined in terms of `lref_major_radius`, and writing back out to GENE (_should_) be idempotent (i.e. not change anything physically), and to convert to other codes we need either a) both `Rmaj` and `a_minor`, or b) the aspect ratio. In practice, we seem to always have `Rmaj` set and only need to supply `a_minor`.

We currently only set `Rmaj` in the Miller geometry from GENE input files, but maybe we should also set `a_minor`? This way, we'd always have the correct aspect ratio used in GENE and so wouldn't need to supply it. How many other codes include the aspect ratio? I believe GS2 can take it as an option (at least two different ways, I'm sure).

I did consider adding the following to `LocalGeometry`:

```diff
 
+    @property
+    def aspect_ratio(self) -> float:
+        return self.Rmaj / self.a_minor
+
+    @aspect_ratio.setter
+    def aspect_ratio(self, value: float):
+        # TODO: Does it make sense to change the aspect ratio if both
+        # R and r are already set?
+        if hasattr(self, "Rmaj"):
+            self.a_minor = self.Rmaj / value
+        elif hasattr(self, "a_minor"):
+            self.Rmaj = self.a_minor * value
+        else:
+            raise RuntimeError(
+                "Cannot set aspect_ratio: missing both minor and major radius"
+            )
+
+    @property
+    def inverse_aspect_ratio(self) -> None:
+        return 1.0 / self.aspect_ratio
+
+    @inverse_aspect_ratio.setter
+    def inverse_aspect_ratio(self, value: float) -> None:
+        self.aspect_ratio = 1.0 / value
+
     # TODO replace this with an abstract classmethod
```